provider/filepath.go

```diff
@@ -0,0 +1,101 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package provider
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	pc "github.com/featureform/provider/provider_config"
+)
+
+type Filepath interface {
+	Bucket() string
+	Prefix() string
+	Path() string
+	FullPathWithBucket() string
+	FullPathWithoutBucket() string
+	ParseFullPath(path string) error
+}
+
+func NewFilepath(storeType pc.FileStoreType, bucket string, prefix string, path string) (Filepath, error) {
+	switch storeType {
+	case S3:
+		return &S3Filepath{
+			filePath: filePath{
+				bucket: strings.Trim(bucket, "/"),
+				prefix: strings.Trim(prefix, "/"),
+				path:   strings.TrimPrefix(path, "/"),
+			},
+		}, nil
+	default:
+		return nil, fmt.Errorf("unknown store type '%s'", storeType)
+	}
+}
+
+type filePath struct {
+	bucket string
```

Comment:
@simba-git, `bucket` is used in the https://pkg.go.dev/gocloud.dev/blob package that we use internal to interact with blob stores. Thus, I wanted to use the same term. But if you think another term is better suited then I am more than happy to change it.